extensions/azurePublishNew/src/node/azureResources/appInsights.ts

```diff
@@ -1,34 +1,138 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-import { ProvisionMethod, ProvisionWorkingSet, ResourceDefinition, ResourceProvisionService } from '../types';
+import { TokenCredentials } from '@azure/ms-rest-js';
+import { ApplicationInsightsManagementClient } from '@azure/arm-appinsights';
+import { AzureBotService } from '@azure/arm-botservice';
 
-import { PAY_AS_YOU_GO_TIER, AZURE_HOSTING_GROUP_NAME } from './constants';
-import { AppInsightsConfigNew } from './types';
+import { ProvisionServiceConfig, ProvisionWorkingSet, ResourceDefinition, ResourceProvisionService } from '../types';
+import { AzureResourceTypes } from '../constants';
+import { AppInsightsResourceConfig } from '../availableResources';
+import {
+  createCustomizeError,
```

Comment:
This method bothers me - I don't understand why we need to manipulate and stringify errors with "owned props".